src/dotnet/Chat.UI.Blazor/Components/ChatNavbarItem.razor

```diff
@@ -0,0 +1,75 @@
+@using BlazorContextMenu
+@inherits ComputedStateComponent<ChatNavbarItem.Model>
+@{
+    var m = State.LatestNonErrorValue;
+}
+
+<ContextMenuTrigger MouseButtonTrigger="MouseButtonTrigger.Right" MenuId="@ChatMenu.MenuId" Data="@Chat" CssClass="flex-x items-center gap-x-2">
+    <NavbarItem Url="@($"/chat/{Chat.Id}")" IsActive="@IsActive">
+        <ChildContent>
+            <div class="flex-x items-center gap-x-2">
+                <ChatIcon Title="@Chat.Title" Picture="@Chat.Picture" Size="SquareSize.Size10" />
+                <span class="text-ellipsis whitespace-nowrap overflow-hidden">@Chat.Title</span>
+            </div>
+        </ChildContent>
+        <Ending>
+            @if (IsListeningPanel) {
+                <ChatListListenToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                @if (m.CanRecord) {
+                    <ChatListRecordingToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                }
+                <ListeningChatRemoveButton ChatId="@Chat.Id"/>
+            } else {
+                <ChatListListenToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                @if (m.CanRecord) {
+                    <ChatListRecordingToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                }
```

Comment:
this part looks duplicate for both `if` and `else` so it can be moved before ifelse